indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexPhaseRunner.java

```diff
@@ -114,6 +115,44 @@
    */
   abstract int getTotalNumSubTasks() throws IOException;
 
+  /**
+   * Workaround for not modifying how {@link org.apache.druid.indexing.common.task.AbstractTask#getOrMakeId} generates
+   * unique IDs (which include the current time).
+   */
+  private class UniqueSubTaskIdDecorator extends ForwardingIterator<SubTaskSpec<SubTaskType>>
+  {
+    private final Iterator<SubTaskSpec<SubTaskType>> delegate;
+
+    UniqueSubTaskIdDecorator(Iterator<SubTaskSpec<SubTaskType>> subTaskSpecIterator)
+    {
+      delegate = subTaskSpecIterator;
+    }
+
+    @Override
+    protected Iterator<SubTaskSpec<SubTaskType>> delegate()
+    {
+      return delegate;
+    }
+
+    @Override
+    public SubTaskSpec<SubTaskType> next()
+    {
+      ensureUniqueSubtaskId();
+      return super.next();
+    }
+
+    private void ensureUniqueSubtaskId()
```

Comment:
Would it be better to save the previous SubTaskSpec and try again if it is the same, rather than forcing a sleep 1ms on each next() call?

Reply:
I'll remove this code since #8612 was merged

Reply:
It's merged now. :slightly_smiling_face: